server/src/main/java/org/elasticsearch/index/query/AbstractQueryBuilderWrapper.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.search.Query;
+import org.elasticsearch.TransportVersion;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+/**
+ * Wrapper for instances of {@link AbstractQueryBuilder} that have been intercepted using the {@link QueryRewriteInterceptor} to
+ * break out of the rewrite phase. These instances are unwrapped on serialization.
+ * @param <T>
+ */
+public class AbstractQueryBuilderWrapper<T extends AbstractQueryBuilder<T>> extends AbstractQueryBuilder<T> {
```

Comment:
nit: It's not an abstract class so let's rename it to InterceptQueryBuilderWrapper and make it a package protected class. It's really an implementation detail and something that should be handled automatically in the `AbstractQueryBuilder#rewrite` implementation.

Reply:
I changed the name based off of [this request](https://github.com/elastic/elasticsearch/pull/117839#discussion_r1873522484), but can change it back.

Reply:
Done in 021d59223aaff950a32cc233ad9e794a642e2a24 FYI @benwtrent